### PR TITLE
fetch CrateDetails via MatchedRelease to save one SQL query per request

### PR DESCRIPTION
**Pull request by syphar**
While working on #2383 I realized that we were fetching all releases for a crate twice, once in `match_version`, once when fetching `CrateDetails`. 

This is the fix, should make some endpoints a little faster: 
- rustdoc pages
- rustdoc redirector
- target redirect
- crate details

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]
Sounds pretty nice! Please wait for @Nemo157's review before merging.

File: src/web/rustdoc.rs
Comment (Nemo157) [inline src/web/rustdoc.rs:735]: We could avoid a bunch of cloning if `CrateDetails::from_matched_release` took the `MatchedRelease` by value, then expose `crate_details.version` to use here.

**Review by Nemo157 — APPROVED** [review]

**Pull request merged**
